src/SceneManager.ts

```diff
@@ -116,6 +131,27 @@ export class SceneManager {
    */
   private audioTopic: string;
 
+  /*
+   * Name of the topic to advertise.
+   */
+  private topicName: string;
+
+  /*
+   * Message type of the topic to advertise.
+   */
+  private msgType: string;
+
+  /*
+   * Message data of the topic to advertise.
+   */
+  private msgData: any
+
+  /*
+   * Publisher object to publish a gazebo message
+   */
+  private publisher: Publisher;
```

Comment:
What's a `Publisher`? I see that this PR point to the `adv_pub` branch. We need a PR of that.

Reply:
Sorry for the confusion. So `adv_pub` is the branch that Nate wrote that allows the client to request the server to advertise a public. I saw that `adv_pub` branch is already 65 commits ahead of main. But I think either Nate or you might be the best person to decide whether we should merge `adv_pub` to main.

And here's the definition for the [Publisher](https://github.com/gazebo-web/gzweb/commit/fc86a123e13c60d125dff296a8deac523c7406ef). This is on the `adv_pub` branch.

Reply:
Yeah, we should not merge it blindly. I'll take a look, but it will take me more time.

Tagging @coleOsrf for awareness

Reply:
Yeah so 
1. We have to cleanup gzweb so that main is somewhat up to date. Cause looking at the branch history, there's been lot of spin offs from `main` but hasn't been merged back. I spun off this PR from Nate's `adv_pub` branch.
2. We could merge this PR to `adv_pub` branch and then merge `adv_pub` to `main` OR first look into merging `adv_pub` into `main` and then merging this PR to `main` as well. Either should be okay.

Reply:
We shouldn't aim to `main` but to `gzweb2` - Once `adv_pub` is up to date with `gzweb2` it will be easier to review, but I repeat, we should _not_ merge it.

Reply:
Okay that sounds good. Let's first merge `adv_pub` to `gzweb2` and then look at this PR. Do you want me to create a PR for merging `adv_pub` to `gzweb2`? And just to be clear, `gzweb2` is the most stable code that we're using for the hyundai project right?

Reply:
Please do not merge `adv_pub` to `gzweb2` - Pull the latest changes of `gzweb2` to `adv_pub` which will make the review process easier.

Reply:
The `adv_pub` branch was updated.

Reply:
> The `adv_pub` branch was updated.

Perfect.  If you open a PR with those changes, I can review them and then we can get that into gzweb2.  Then we can rebase this one to point directly to gzweb2.

Reply:
The PR is open: https://github.com/gazebo-web/gzweb/pull/29 Once it's merged, this branch will automatically point to `gzweb2`.